src/content/en/fundamentals/engage-and-retain/push-notifications/subscribing-a-user.md

```diff
@@ -0,0 +1,396 @@
+message.](./images/svgs/application-server-key-send.svg)
+
+Technically, the `applicationServerKey` is optional. However, the easiest
+implementation on Chrome requires it and other browsers may require it in
+the future. It's optional on Firefox.
+
+The specification that defines *what* the application server key should be is
+the [VAPID spec](https://tools.ietf.org/html/draft-thomson-webpush-vapid).
+Whenever you read something referring to *"application server keys"* or
+*"VAPID keys"*, just remember that they are the same thing.
+
+Those are the only subscribe option.
+
+#### How to Create Application Server Keys
+
+You can create a public and private set of application server keys by visiting this web site
+[web-push-codelab.appspot.com](https://web-push-codelab.appspot.com/) or you can use the
+[web-push](https://github.com/web-push-libs/web-push#command-line) to generate keys by doing
+the following:
+
+    $ npm install -g web-push
+    $ web-push generate-vapid-keys
+
+You only need to create these keys once for your application, just make sure you keep the
+private key private....yeah I just said that.
+
+## Permisions and subscribe()
+
+There is one side effect of calling `subscribe()`. If your web app doesn't have permissions for
+showing notifications at the time of calling `subscribe()`, the browser will request the
+permissions for you. This is useful if your UI works with this flow, but if you want more
+control (and I think most developers will), stick to the `Notification.requestPermission()` API
+that we used earlier.
+
+## What is a PushSubscription?
+
+We call `subscribe()`, pass in some options, in return we get a promise that resolves to a
+`PushSubscription` resulting in some code like so:
+
+    function subscribeUserToPush() {
+      return getSWRegistration()
+      .then(function(registration) {
+        const subscribeOptions = {
+          userVisibleOnly: true,
+          applicationServerKey: urlBase64ToUint8Array(
+            'BEl62iUYgUivxIkv69yViEuiBIa-Ib9-SkvMeAtA3LFgDzkrxZJjSgSnfckjBJuBkr3qBUYIHBQFLXYp5Nksh8U'
+          )
+        };
+
+        return registration.pushManager.subscribe(subscribeOptions);
+      })
+      .then(function(pushSubscription) {
+        console.log('Received PushSubscription: ', JSON.stringify(pushSubscription));
+        return pushSubscription;
+      });
+    }
+
+The `PushSubscription` object contains all the required information needed to send a push
+messages to that user. If you print out the contents using `JSON.stringify()`, you'll see the
+following:
+
+    {
+      "endpoint": "https://some.pushservice.com/something-unique",
+      "keys": {
+        "p256dh":
+    "BIPUL12DLfytvTajnryr2PRdAgXS3HGKiLqndGcJGabyhHheJYlNGCeXl1dn18gSJ1WAkAPIxr4gK0_dQds4yiI=",
+        "auth":"FPssNDTKnInHVndSTdbKFw=="
+      }
+    }
+
+The `endpoint` is the push services URL, you make a POST request to this URL to trigger a push
```

Comment:
Change 'you make a POST request' to 'your server makes a POST request'. Trying to avoid POV confusion.